packages/client/src/runtime/core/engines/library/ExitHooks.ts

```diff
@@ -15,11 +16,12 @@ export class ExitHooks {
       return
     }
 
-    this.installHook('beforeExit')
-    this.installHook('exit')
-    this.installHook('SIGINT', true)
-    this.installHook('SIGUSR2', true)
-    this.installHook('SIGTERM', true)
+    process.once('exit', () => this.triggerAllListeners())
```

Comment:
Not needed after #18848 is merged, TS just forced me to fix the same issue as well